### ISSUE TITLE
Problem verifying container signed by GH actions

### ISSUE THREAD

**Issue by JamieW87** [sigstore/cosign#2640]
**Description**

I have introduced Cosign into my GH Actions workflow to sign my docker containers.

Currently having an issue verifying the signatures on the images that have been signed during the workflow.

I generate the key pair. Put the prv in Actions secrets and keep the pub locally.
I export the COSIGN_PASSWORD env var.

In the actions workflow, after the docker image is pushed and tagged, I sign it with:

```
cosign sign -key cosign.key example/name:0.6.5
```

This works fine and the expected signature turns up in Docker hub.
However when i run the verifiy command:

```
cosign verify --key cosign.pub example/name@sha256:56b3d24bcedd9b2ed891d529b6055bfc34ec7d7a77681326470f8ee7a7a4e34e
```

I get this error:
```
Error: no matching signatures:
searching log query: [POST /api/v1/log/entries/retrieve][404] searchLogQuery default  &{Code:404 Message:no responses found}
main.go:63: error during command execution: no matching signatures:
searching log query: [POST /api/v1/log/entries/retrieve][404] searchLogQuery default  &{Code:404 Message:no responses found}
```

**HOWEVER**

If I run both the exact same commands locally, the verify step works as expected.
I have CATted the key out into the workflow to ensure its the same one, and the key pair are identical.

Am i missing some context here?

**Version**

Using the `run sigstore/cosign-installer@main` command in the workflow, which is giving me v1.13.1

Locally: cosign version prints this:
```
GitVersion:    devel
GitCommit:     e69e19027327c45057a0d3636d65f0d4c1374638
GitTreeState:  clean
BuildDate:     2023-01-17T18:21:08
GoVersion:     go1.18.1
Compiler:      gc
Platform:      darwin/amd64
```

**Comment by znewman01** [comment]
We discussed this in Slack. Turns out the root issue was that GH Actions uses the 1.13.1 Cosign release, in which transparency log upload is experimental, so you'd need `COSIGN_EXPERIMENTAL=1`. In the 2.x release candidates, the need for `COSIGN_EXPERIMENTAL` is removed; hopefully a full 2.0 release goes out soon so this will be totally obsolete.

That *should* resolve the issue, so I'm closing. However, if it doesn't, let me know and I'd be happy to reopen and continue troubleshooting.
